### DISCUSSION TITLE
no_std environment for meilisearch

### DISCUSSION THREAD

**Discussion by silicology1** [meilisearch/product#698] [Feedback & Feature Proposal]
Using meilisearch in blockchain has many use cases, like indexing article feeds of data stored in blockchain, or using it for any search engines without relying on centralized servers. Like building decentralized google,  for indexing different servers of mastodon for better search which is not possible now.  Substrate blockchain provides offchain computing using offchain worker. But offchain worker only supports [no_std](https://docs.substrate.io/reference/how-to-guides/offchain-workers/offchain-http-requests/). Also substrate [sidechains](https://github.com/integritee-network/integritee-node/issues/195#issuecomment-1752803435) can be used for more scalability. meilsearch can provide a no_std crate to be used with substrate blockchains. 

Another needed feature is ability to store meilisearch data in [ipfs/iroh](https://github.com/n0-computer/iroh)

**Comment by macraig** [comment]
Thank you for sharing your use case @amiyatulu !

Currently, our team is focused on other prioritized initiatives. However, your feedback is valuable and we’re keeping the discussion open for other users to comment and upvote on our platform. Your voice matters and it helps shape our future developments. We appreciate your understanding and continued support!
